From 896d77dd949b52d6ea5798e5a038ba97d6b802be Mon Sep 17 00:00:00 2001
From: Peter Marko <peter.marko@siemens.com>
Date: Thu, 4 Apr 2024 15:44:18 +0200
Subject: [PATCH] macros: guard ipv6 code with SYSLOG_NG_ENABLE_IPV6

With ipv6 disabled, there are linking errors currently.
This fixes it by not using the symbols when IPv6 is disabled.

Solves #4810 with my config options
https://github.com/openembedded/meta-openembedded/blob/2487e65ee3842b6ae0c7a2628985be6189ed9ebf/meta-oe/recipes-support/syslog-ng/syslog-ng_4.6.0.bb

Upstream-Status: Submitted [https://github.com/syslog-ng/syslog-ng/pull/4880]

Signed-off-by: Peter Marko <peter.marko@siemens.com>
---
 lib/template/macros.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/template/macros.c b/lib/template/macros.c
index 54142706c..fcb83637f 100644
--- a/lib/template/macros.c
+++ b/lib/template/macros.c
@@ -277,12 +277,14 @@ _get_originating_ip_protocol(const LogMessage *msg)
     return 0;
   if (g_sockaddr_inet_check(msg->saddr))
     return 4;
+#if SYSLOG_NG_ENABLE_IPV6
   if (g_sockaddr_inet6_check(msg->saddr))
     {
       if (g_sockaddr_inet6_is_v4_mapped(msg->saddr))
         return 4;
       return 6;
     }
+#endif
   return 0;
 }
 
-- 
2.30.2

